src/game/stage/item_box.c

```diff
@@ -0,0 +1,518 @@
+                }
+            }
+
+        } break;
+
+        case 2: {
+            if (!(gPlayer.unk37 & 0x8)) {
+                gPlayer.unk37 &= ~0x1;
+                gPlayer.unk37 |= 0x8;
+
+                if (!(gPlayer.unk37 & 0x2)) {
+                    sub_802A8F8(gPlayer.unk60);
+                }
+            }
+
+        } break;
+
+        case 3: {
+            gPlayer.unk2E = 0x4B0;
+
+            if (IS_SINGLE_PLAYER || !(gPlayer.unk37 & 0x2)) {
+                gPlayer.unk37 |= 0x2;
+                sub_802A854(gPlayer.unk60);
+                gUnknown_030054A8.unk2 = 16;
+            }
+        } break;
+
+        case 4: {
+            gPlayer.unk37 |= 0x4;
+            gPlayer.unk30 = 0x4B0;
+
+            m4aMPlayTempoControl(&gMPlayInfo_BGM, 0x200);
+
+            if (IS_MULTI_PLAYER) {
+                gPlayer.unk37 &= ~0x10;
+            }
+        } break;
+
+        case 5: {
+            s32 rnd = gUnknown_080D51FC[(u32)PseudoRandom32() % 5];
+            rings = &gRingCount;
+            oldRingCount = *rings;
+            newRingCount = *rings + rnd;
+            goto ApplyItemboxEffect_case7;
+        } break;
+
+        case 6: {
+            rings = &gRingCount;
+            oldRingCount = *rings;
+            newRingCount = *rings + 5;
+            goto ApplyItemboxEffect_case7;
+        } break;
+
+        case 7: {
+            rings = &gRingCount;
+            oldRingCount = *rings;
+            newRingCount = *rings + 10;
+
+        ApplyItemboxEffect_case7:
+            gRingCount = newRingCount;
+            if (gCurrentLevel != LEVEL_INDEX(ZONE_FINAL, ACT_TRUE_AREA_53)) {
+                s32 newLivesCount = Div(gRingCount, 100);
+
+                if (newLivesCount != Div(oldRingCount, 100)
+                    && (gGameMode == GAME_MODE_SINGLE_PLAYER)) {
+                    u32 newLives = gNumLives;
+                    if (++newLives > 255) {
+                        newLives = 255;
+                    }
+                    gNumLives = newLives;
+
+                    gUnknown_030054A8.unk3 = 16;
+                }
+            }
+
+            if (gGameMode == GAME_MODE_MULTI_PLAYER_COLLECT_RINGS) {
+                if (gRingCount > 255) {
+                    gRingCount = 255;
+                }
+            }
+
+            m4aSongNumStart(SE_RING_COPY);
+
+        } break;
+
+        case 8: {
+            s32 smallestMagnitude = -1;
+            u8 nearestPlayer = 0;
+            u8 playerId;
+
+            // Find the player that's closest to you
+            for (playerId = 0; playerId < MULTI_SIO_PLAYERS_MAX
+                 && gMultiplayerPlayerTasks[playerId] != NULL;
+                 playerId++) {
+                struct MultiplayerPlayer *mpp;
+                s32 boxToPlayerX, boxToPlayerY;
+                s32 boxToPlayerMagnitude;
+
+                // Don't look for your own ID
+                u32 sioId = (*(vu32 *)&REG_SIOCNT << 26) >> 30;
```

Comment:
There is a macro already for this `(SIO_MULTI_CNT)->id;`

Reply:
Awesome!